src/data/user-service.ts

```diff
@@ -0,0 +1,95 @@
+import { gql, useLazyQuery, useMutation } from '@apollo/client';
+import { UserRoleField } from '../components/form-fields';
+import { FieldErrors } from './errors';
+import { navigateToHome, returnPage } from './navigation-service';
+
+export enum UserRole {
+  admin = 'admin',
+  user = 'user',
+}
+
+export interface UserInterface{
+  email: string | undefined,
+  name: string | undefined,
+  phone: string | undefined,
+  birthDate: string | undefined,
+  role: UserRole | undefined
+}
+
+export interface UserErrorInterface{
+  email: FieldErrors | undefined,
```

Comment:
Não daria o mesmo de colocar o campo como opcional? 
```suggestion
  email?: FieldErrors,
```

Reply:
Justo, eu não conhecia essa sintaxe